dashboard/app/views/levels/instructions.haml

```diff
@@ -0,0 +1,26 @@
+-#
+  This is a view for rendering a summary of all of the levels/instructions in a given
+  script, that should only be visible in levelbuilder mode
+
+- stages = local_assigns[:stages]
+
+- stages.each_with_index do |stage, index|
+  %h1{:style => "margin-top: 30px"}= "Stage #{(index + 1)}: #{stage.name}"
+  - stage.script_levels.each do |script_level|
+    - level = script_level.level
+    %div{:style => "border: 1px dashed black; padding: 10px"}
+      = link_to "Level #{script_level.position}: #{script_level.level.name}", build_script_level_path(script_level)
+      %div= "type: #{script_level.level.class}"
+      - levels = level.is_a?(LevelGroup) ? level.levels : [level]
+      - levels.each do |level|
+        - if level.class == External
```

Comment:
One middling approach might be to have level types optionally specify a to_instructions method, and we call that if it exists, otherwise we do our own thing. This way, the fact that External is unique becomes a little bit more generalized, but we dont have to override for every level type

i.e. instead of `if level.class == External` we would do `return level.to_instructions if level.responds_to?(:to_instructions)`